crates/bevy_ecs/src/hierarchy.rs

```diff
@@ -96,6 +98,11 @@ use log::warn;
     feature = "bevy_reflect",
     reflect(Component, PartialEq, Debug, FromWorld, Clone)
 )]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(
+    all(feature = "serialize", feature = "bevy_reflect"),
+    reflect(Serialize, Deserialize)
+)]
 #[relationship(relationship_target = Children)]
 #[doc(alias = "IsChild", alias = "Parent")]
 pub struct ChildOf(pub Entity);
```

Comment:
Ah, you need to explicitly mark entity:
```suggestion
pub struct ChildOf(#[entities] pub Entity);
```